ksonnet-gen/ksonnet/emit.go

```diff
@@ -422,6 +429,20 @@ func (ao *apiObject) emitAsRefMixins(
 	m.writeLine("},")
 }
 
+func (ao *apiObject) emitConstructor(m *indentWriter) {
+	if dm, ok := ao.propertyMethods["default"]; ok {
+		log.Fatalf(
+			"Attempted to create constructor, but 'default' property already existed at '%s'",
+			dm.path)
+	}
+
+	if _, ok := ao.propertyMethods["metadata"]; !ok {
+		m.writeLine("default():: apiVersion + kind + {},")
+	} else {
+		m.writeLine("default(name):: apiVersion + kind + self.mixin.metadata.name(name),")
```

Comment:
SGTM -- let me know if you want to merge now or later.  Feel free to self merge now.